curriculum/test/test-challenges.js

```diff
@@ -86,59 +88,141 @@ spinner.text = 'Populate tests.';
 let browser;
 let page;
 
-runTests();
-
-async function runTests() {
-  process.on('unhandledRejection', err => {
-    spinner.stop();
-    throw new Error(`unhandledRejection: ${err.name}, ${err.message}`);
+setup()
+  .then(runTests)
+  .catch(err => {
+    cleanup();
+    // setting the error code because node does not (yet) exit with a non-zero
+    // code on unhandled exceptions.
+    process.exitCode = 1;
+    throw err;
   });
 
-  const testLangs = testedLangs();
+async function setup() {
+  if (process.env.npm_config_superblock && process.env.npm_config_block) {
+    throw new Error(`Please do not use both a block and superblock as input.`);
+  }
 
-  const challenges = await Promise.all(
+  // liveServer starts synchronously
+  liveServer.start({
+    host: '127.0.0.1',
+    port: '8080',
+    root: path.resolve(__dirname, 'stubs'),
+    mount: [['/js', path.join(clientPath, 'static/js')]],
+    open: false,
+    logLevel: 0
+  });
+  browser = await puppeteer.launch({
+    args: [
+      // Required for Docker version of Puppeteer
+      '--no-sandbox',
+      '--disable-setuid-sandbox',
+      // This will write shared memory files into /tmp instead of /dev/shm,
+      // because Docker’s default for /dev/shm is 64MB
+      '--disable-dev-shm-usage'
+      // dumpio: true
+    ]
+  });
+  global.Worker = createPseudoWorker(await newPageContext(browser));
+  page = await newPageContext(browser);
+  await page.setViewport({ width: 300, height: 150 });
+  const testLangs = testedLangs();
```

Comment:
```suggestion
  const testLangs = testedLangs();
  if (testLangs.length > 1)
    throw Error(
      `Testing more than one language at once is not currently supported
please change the TEST_CHALLENGES_FOR_LANGS env variable to a single language`
    );
```
We may not need to support multiple languages, but we should tell people why it fails.